decoder/src/log/format.rs

```diff
@@ -1,48 +1,233 @@
+//! Formatting for `defmt` log frames.
```

Comment:
Thank you for writing this specification. I think that we should make it publicly accessible, which it is not atm, since the `format` module is private. Either we add it to the [defmt book](https://defmt.ferrous-systems.com), or to the `log` module. What do you think?

Reply:
I was looking into adding this into the book too. I also realized after I opened this PR that rustdoc doesn't include this when building documentation since it's a private module.

Reply:
If you don't mind, I can do the book thing in a separate PR.

Reply:
Yes, that sounds good